com.unity.render-pipelines.universal/ShaderLibrary/Lighting.hlsl

```diff
@@ -230,9 +231,10 @@ int GetPerObjectLightIndex(uint index)
     // We limit to 4 indices per object and only sample unity_4LightIndices0.
     // Conditional moves are branch free even on mali-400
     // small arithmetic cost but no extra register pressure from ImmCB_0_0_0 matrix.
-    half2 lightIndex2 = (index < 2.0h) ? unity_LightIndices[0].xy : unity_LightIndices[0].zw;
-    half i_rem = (index < 2.0h) ? index : index - 2.0h;
-    return (i_rem < 1.0h) ? lightIndex2.x : lightIndex2.y;
+    half indexHalf = half(index);
+    half2 lightIndex2 = (indexHalf < half(2.0)) ? unity_LightIndices[0].xy : unity_LightIndices[0].zw;
```

Comment:
I wonder - is unity_LightIndices half already?

Reply:
No. It was defined as real.
I've changed that now to half and fixed the functions that used it.